### PR TITLE
Gitclone

### PR DESCRIPTION
**Pull request by ericfode**

### CODE REVIEW

File: lib/ferret.rb [hunk 1]
```diff
@@ -145,6 +145,7 @@ def test(opts={}, &blk)
   rescue Timeout::Error
     log source: source, at: :timeout, val: opts[:timeout]
     if opts[:pid]
+      puts "timout"
```
Comment (nzoschke) [inline lib/ferret.rb]: typo and unneeded debugging...

File: tests/git_clone [hunk 2]
```diff
@@ -2,15 +2,22 @@
 
 require_relative "../lib/ferret"
 
-bash(retry: 2, name: :heroku_info_create, stdin: <<'EOF')
-  heroku info --app $TARGET_APP || {
-    heroku create $TARGET_APP                                             \
-      && heroku plugins:install https://github.com/heroku/manager-cli.git \
-      && heroku manager:transfer --app $TARGET_APP --to $ORG
-  }
+bash(retry: 1, name: :setup, stdin: <<'EOF')
```
Comment (nzoschke) [inline tests/git_clone]: This doesn't need any code pushed, just an app record created.

I think options to uses_app would be nice, one for stack and make the code directory optional. Then we could re-use the helper in this test too.

File: Gemfile [hunk 3]
```diff
@@ -1,3 +1,4 @@
 source :rubygems
 gem 'sinatra'
-gem 'thin'
\ No newline at end of file
+gem 'thin'
+gem 'sequel'
```
Comment (nzoschke) [inline Gemfile:4]: does ferret itself need these dependencies?

Comment (ericfode) [inline Gemfile:4]: No The apps in the app directory do, and the test for http_warmup does

**Pull request merged**